src/Symfony/Component/Translation/Translator.php

```diff
@@ -72,32 +73,53 @@ class Translator implements TranslatorInterface, TranslatorBagInterface
     private $configCacheFactory;
 
     /**
-     * Constructor.
-     *
-     * @param string               $locale   The locale
-     * @param MessageSelector|null $selector The message selector for pluralization
-     * @param string|null          $cacheDir The directory to use for the cache
-     * @param bool                 $debug    Use cache in debug mode ?
+     * @var CatalogueBagInterface
+     */
+    private $translatorBag;
+
+    /**
+     * @param string                                     $locale        The locale
+     * @param CatalogueBagInterface|MessageSelector|null $translatorBag The CatalogueBagInterface or MessageSelector
+     * @param MessageSelector|string|null                $selector      The MessageSelector or cache directory
+     * @param bool                                       $debug         Use cache in debug mode ?
+     *                                                                  Deprecated since version 2.8, to be removed in 3.0.
      *
      * @throws \InvalidArgumentException If a locale contains invalid characters
      *
      * @api
      */
-    public function __construct($locale, MessageSelector $selector = null, $cacheDir = null, $debug = false)
+    public function __construct($locale, $translatorBag = null, $selector = null, $debug = false)
```

Comment:
fixed and here is how it was solved https://github.com/aitboudad/symfony/blob/translation_loader_resolver/src/Symfony/Component/Translation/Translator.php#L303-L324